packages/ui/src/InteractiveList.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { FocusEvent, useCallback, useMemo, useState } from 'react'
+import { TextField, TextFieldExtraProps } from './TextField'
+import { IconButton } from './IconButton'
+import { Plus, X } from 'react-feather'
+import styles from './InteractiveList.module.scss'
+import { useUID } from 'react-uid'
+import { FieldArrayRenderProps, FieldValidator } from 'formik'
+
+export type InteractiveListExtraProps = {
+  customAddElement?: React.FC<{ setValue: (value: string) => void }>
+} & Pick<TextFieldExtraProps<'text'>, 'label' | 'helperText' | 'inputIcon' | 'type'>
+
+export type InteractiveListCoreProps<V> = {
+  name: string
+  value: string[]
+  onBlur?: (e: FocusEvent<HTMLTextAreaElement>) => void
+  error?: string
+  validate?: FieldValidator
+  arrayHelpers: FieldArrayRenderProps
+  onError: (str: string) => void
+} & InteractiveListExtraProps
+
+export type InteractiveListProps<V> = InteractiveListExtraProps & InteractiveListCoreProps<V>
+
+const InteractiveList = <V extends object>({
+  label,
+  error,
+  name,
+  customAddElement: CustomAddElement,
+  value,
+  helperText,
+  inputIcon,
+  onError,
+  type,
+  validate,
+  arrayHelpers,
+}: InteractiveListProps<V>) => {
+  const id = useUID()
+  const [inputValue, setValue] = useState<string>('')
+
+  const normalizedValue = useMemo(() => inputValue.trim(), [inputValue])
+
+  const addValueCallback = useCallback(async () => {
+    if (normalizedValue.length === 0) {
+      onError('You need to provide a non empty value')
+    } else if (value.includes(normalizedValue)) {
+      onError('You need to provide a unique value')
+    } else {
+      if (validate) {
+        const error = await validate(normalizedValue)
+        if (error) {
+          onError(error)
+        } else {
+          arrayHelpers.push(normalizedValue)
+          setValue('')
+        }
+      } else {
+        arrayHelpers.push(normalizedValue)
+        setValue('')
+      }
+    }
+  }, [value, normalizedValue])
+
+  return (
+    <>
+      <div className={styles.inputRow}>
+        <TextField
```

Comment:
@jvorcak I generally like the idea, but I am wondering if we could make the validation function totally custom. I would suggest to have 2 props:
- `validateCommitted` that validates the already added values in formik. It could be used to validate the array values.
- `validateUncommitted` that validates the current input value and receives the input value and the current array. It does not block the submission.

Reply:
@aigoncharov I like the way where you want to go.

Basically `validateCommited` will not be needed, it'll be the default Formik validation (that will also work with `Yup` if present), that is how validation for `<FieldArray />` is meant - more about the problematic here -> https://formik.org/docs/api/fieldarray#fieldarray-validation-gotchas

The `validateUncommited` will be our custom function that will be used as a reaction to `<TextField/>` change, that is something Formik's FieldArray doesn't count on. WDYT?

Reply:
@jvorcak we'll still need to pass the default formik validation to the formik-bound field, right? So `validateCommited` is meant for `InterqctiveListFormik`.

Reply:
@aigoncharov btw, what is the advantage of having a separate validation for the current input? Why can't we just use the `validate` that is passed to a Formik's `useField`?

Reply:
@jvorcak good idea! Let's do exactly that!